library/core/src/ptr/mut_ptr.rs

```diff
@@ -1548,12 +1548,13 @@ impl<T: ?Sized> *mut T {
     /// # fn foo(n: usize) {
     /// # use std::mem::align_of;
     /// # unsafe {
-    /// let x = [5u8, 6u8, 7u8, 8u8, 9u8];
-    /// let ptr = x.as_ptr().add(n) as *const u8;
+    /// let mut x = [5u8, 6, 7, 8, 9];
+    /// let ptr = x.as_mut_ptr().add(n);
     /// let offset = ptr.align_offset(align_of::<u16>());
     /// if offset < x.len() - n - 1 {
-    ///     let u16_ptr = ptr.add(offset) as *const u16;
+    ///     let u16_ptr = ptr.add(offset) as *mut u16;
     ///     assert_ne!(*u16_ptr, 500);
+    ///     *u16_ptr = 0;
```

Comment:
With this doing an update, it would be nice, I think, to have a validation of some sort *showing* what that did.

Is it `assert!(x == [0, 0, 7, 8, 9] || x == [5, 0, 0, 8, 9]);`?

(Similarly for the `const` version -- maybe check that it read one `0x0506.from_le()` or `0x0607.from_le()`?)

Reply:
The problem is that this code is not even tested (`fn foo(n: usize) {`) 🤔 

Maybe we can remove the function and `.add(n)`?

Reply:
Also it should be `from_be` :P

these methods are so confusing...

Reply:
Agreed -- I should have written it using the `_bytes` versions, like you did in the update.  The `uNN` <--> `uNN` versions are evil.